backend/src/openai.ts

```diff
@@ -271,8 +271,8 @@ async function chatGptChatCompletion(
 
 		// check to see if there's already a system role
 		const systemRole = chatHistory.find(
-			(message) => message.completion?.role === 'system'
-		);
+			(message) => message.chatMessageType === CHAT_MESSAGE_TYPE.SYSTEM
+		) as ChatSystemMessage | undefined;
```

Comment:
We need this cast. Without it, the compiler does not infer that the `systemRole` will have a `.completion`, and there is an error when when we try to access it on line 284 (in the else statement a few lines down)